### DISCUSSION TITLE
使用 Module federation 对组件开发的限制

### DISCUSSION THREAD

**Discussion by weieyuan** [empjs/emp#102] [General]
如果我有两个前端项目 A 和 B，两个项目可以独立部署和访问，例如 A 项目的地址为 https://a.123.com；B 项目的地址为 https://b.456.com。

这个时候如果 A 项目需要使用 B 项目的组件 SomeComponent，但是 SomeComponent 中可能封装了请求，可能使用了全局的数据，那么当在 A 项目中加载使用 SomeComponent 的时候，可能会存在诸多的问题（比如请求会跨域），那么需要对 SomeComponent 进行重新的封装。

那如果 A 项目 和 B 项目是不同团队开发维护的，怎么确保 B 项目中的 SomeComponent 在后续的迭代中，不会影响在 A 项目中的正常使用了？

核心有两个问题：
* 为了使用 Module federation 的功能，需要对”共享“组件进行重新封装，会存在迁移的成本
* 后续的维护成本

**Comment by ckken** [comment]
+ 交集太大建议独立 base基站
+ 维护成本问题可以考虑停用 unpkg 做多版本管理(按项目需求)
